src/cli.rs

```diff
@@ -269,9 +272,25 @@ fn get_src_bin_tree_sitter_lint_local_lsp_rs_contents(local_binary_crate_name: &
                     .with_env_filter(EnvFilter::from_default_env())
                     .with_writer(out_log)
                     .init();
-            }
+                None
+            } else {
+                None
+            };
 
-            #local_binary_crate_name::run_lsp().await;
+            // FlushGuard isn't Sync so can't just pass it
+            // around directly I guess (for being able to end/start
+            // new traces)
+            let sender = guard.map(|guard| {
```

Comment:
This is what I came up with to get around the fact that I guess `FlushGuard`'s aren't `Sync` so couldn't easily get shared into the LSP server async stuff (so basically if we're in "dump trace files" mode, start up a "background" thread that just listens for "start a new trace" messages)